### PR TITLE
Updating astropy-helpers to latest v2.0dev

### PR DESCRIPTION
**Pull request by bsipocz**
~~DO NOT MERGE! This is a testing, placeholder PR until ah is released.~~

EDIT: this PR now contains only the helpers update to 2.0rc1 and related docs changes. Can be merged once CI passes.

There were quite a few changes in the helpers, and thus I see only benefits of trying to test against the latest dev branch well before the feature freeze.

E.g. I expect a few docs issues with the new sphinx version, but we need this helpers to be able to test it.

### CODE REVIEW

**Comment by bsipocz** [comment]
I'm not sure where is this undefined reference:
https://travis-ci.org/astropy/astropy/jobs/238946509#L4481

The other warnings seem to be due to #4927 and thus will need to be fixed before the release.

**Comment by bsipocz** [comment]
I think the helpers is now in a working state, but nevertheless, we need to sort out the docs warnings before having the release. Currently there are 11 of them, mostly related to #6232 and #4927.

**Comment by pllim** [comment]
See sphinx-gallery/sphinx-gallery#255 for potential fix to the `RemovedInSphinx17Warning` in test log.

Also see bsipocz/astropy#2 for potential fix to "Ripley footnote not referenced" warnings.

**Comment by pllim** [comment]
Update for today: I removed my Ripley ref fix in favor of #6252 and rebased against latest master.

**Comment by bsipocz** [comment]
@pllim - Don't worry about the commit history here, at the end of the day all the sphinx fixing commit should go into another PR, and this will only contain the astropy helpers update. But currently it's more convenient to test the former here, too as it needs the fixes from the helpers.

**Comment by pllim** [comment]
@bsipocz , I hunted down the R115 warning, which is caused by wrong indentation at https://github.com/astropy/astropy/blob/master/astropy/modeling/optimizers.py#L187 !

**Comment by pllim** [comment]
Update: I understand all the warnings now. Some are fixed here and some need external fixes. See https://github.com/astropy/astropy/pull/6140#issuecomment-309882840

**Comment by bsipocz** [comment]
@pllim - I've restarted the build now that the data file updates are merged.

**Comment by pllim** [comment]
OMG it's **green** -- https://travis-ci.org/astropy/astropy/jobs/245655085

**Comment by pllim** [comment]
@bsipocz , you said the warning fixes should go in a different PR, but how? Are you going to cherry pick, or am I going to copy-and-paste the code into a new branch?

File: .travis.yml [hunk 1]
```diff
@@ -144,6 +145,10 @@ install:
     - git clone git://github.com/astropy/ci-helpers.git
     - source ci-helpers/travis/setup_conda.sh
 
+    # Use this until https://github.com/sphinx-gallery/sphinx-gallery/pull/255
+    # is merged.
+    - if [[ $SETUP_CMD == *build_docs* ]]; then pip -q install git+https://github.com/pllim/sphinx-gallery.git@stat-iter-deprec#egg=sphinx-gallery; fi
```
Comment (pllim) [inline .travis.yml]: Note: This fixes the deprecation warning from Sphinx for sure. This hack will have to stay until my PR over at `sphinx-gallery` gets merged, which I have no control over. But feel free to suggest a more elegant hack.

Comment (bsipocz) [inline .travis.yml]: does this warning cause a sphinx failure? if it does, could we maybe use the functionality of https://github.com/astropy/astropy/pull/6223 to turn it off so not just travis, but local builds would pass without issues?

**Comment by bsipocz** [comment]
@pllim - I planned to cherry-pick them, but let's have the helpers RC out first, so we don't just do a blind merge, but test it before that.

File: .travis.yml (hunk 1 above)
Comment (pllim) [inline .travis.yml]: Actually I am not sure. The test log reported failure and the total number of warnings but did not say *which* ones counted. Should I undo this and see what happens?

**Comment by pllim** [comment]
> I planned to cherry-pick them

Okay, I'll leave you to your magic then. :smile:

**Comment by bsipocz** [comment]
OK, so now I've removed all the sphinx fixes from here to #6272, and this PR goes back to test the helpers RC only.

**Comment by bsipocz** [comment]
Merging this now with the helpers 2.0rc1, but we need to update to 2.0 before the release.

**Pull request merged**